event/kefka_tower.py

```diff
@@ -248,16 +248,66 @@ def atma_battle_mod(self):
             field.InvokeBattle(boss_pack_id),
         )
 
+    # Copy no less than 4 bytes between start_target and end_target
+    # This will be called after one of the kt encounters has completed, but just prior to finishing the check
+    def kt_encounter_objective_mod(self, boss_name, bit, start_target, end_target, description):
+        src = Read(start_target, end_target)
+        src += [
+            field.SetEventBit(bit),
+            field.CheckObjectives(),
+            field.Return(),
+        ]
+        post_battle = Write(Bank['CC'], src, f"{boss_name} post-battle. 1) Set event bit. 2) Finish check")
+
+        space = Reserve(start_target, end_target, description, asm.NOP())
+        space.write([
+            field.Call(post_battle.start_address)
+        ])
+
     def guardian_mod(self):
-        pass
+        self.kt_encounter_objective_mod(
+            "Guardian",
+            event_bit.DEFEATED_GUARDIAN,
+            0xc186c,
```

Comment:
This causes all of these event commands to be replaced:

```
CC/186C: 96    Restore screen from fade
CC/186D: 5C    Pause execution until fade in or fade out is complete
CC/186E: D0    Set event bit $1E80($0BC) [$1E97, bit 4]
```

Is that what's desired or should the 96 and 5C be included in the new code? (similar comment on inferno)